### PR TITLE
APIClient WIP

### PR DESCRIPTION
**Pull request by Jdville03**
Here is the first draft for APIClient (REST API). It's really simple as we discussed. I left placeholders if we want to include examples for requests and responses. It wasn't clear to me whether some of the methods had responses so not all of them have them.

Feel free to edit directly if easier or you can also provide comments and i'll go back and fix things you find.

- [x] Structure and organization of methods for Openlaw.scala
- [x] Fill in TODO descriptions
- [x] Examples

### CODE REVIEW

File: docs/api-client/README.md [hunk 1]
```diff
@@ -0,0 +1,929 @@
...
+
+Example
+
+```
+
+```
+
+### uploadContractToGoogle
+
+Upload a contract to user's Google Drive.
```
Comment (outkaj) [inline docs/api-client/README.md]: This feature is not merged yet pending some UX fixes, should we leave this out for now?

File: docs/api-client/README.md [hunk 2]
```diff
@@ -0,0 +1,929 @@
...
+
+```
+
+```
+
+## User
+
+### getUserDetails
+
+Method used as part of `IdentityVariable` to get information about a user.
```
Comment (outkaj) [inline docs/api-client/README.md]: Let's explain more what an `IdentityVariable` is

File: docs/api-client/README.md [hunk 3]
```diff
@@ -0,0 +1,929 @@
...
+
+Example
+
+```
+{
+
+}
+```
+
+### getStripeAccounts
```
Comment (outkaj) [inline docs/api-client/README.md]: Let's leave this out until the Stripe integration is complete/merged.

File: docs/api-client/README.md [hunk 4]
```diff
@@ -0,0 +1,929 @@
...
+
+Example
+
+```
+
+```
+
+### toRestricted
+
+Change role of a user to `NoAccessUser`.
```
Comment (outkaj) [inline docs/api-client/README.md]: We should briefly clarify that `NoAccess` is the same as `Restricted`.

File: docs/api-client/README.md [hunk 5]
```diff
@@ -0,0 +1,929 @@
...
+{
+
+}
+```
+
+## Address
+
+### getAddressDetails
+
+Method used as part of `AddressVariable` to get details about a selected address.
```
Comment (outkaj) [inline docs/api-client/README.md]: We should also clarify a bit more about the `AddressVariable` type.

File: docs/api-client/README.md [hunk 6]
```diff
@@ -0,0 +1,929 @@
...
+
+**Parameters**
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| `title` | `string` | **Required.** The title of the template. |
+
+Example
+
+```
```
Comment (outkaj) [inline docs/api-client/README.md]: @adridadou what do you think is the best way to work in the examples? (Before merging this, I think we should probably either remove the example category or include them).

File: docs/api-client/README.md [hunk 7]
```diff
@@ -0,0 +1,929 @@
+# REST API
+
+The OpenLaw REST API in APIClient.js is an interface for querying, saving, and changing data in an OpenLaw instance. The API methods are categorized below.
```
[the remark this answers is not in the document]
Comment (Jdville03) [reply docs/api-client/README.md]: For the first point, are we assuming that the appropriate code will be open sourced in conjunction with the release of this documentation? Given the context that David provided re developers will have access to code and these APIs to build around.

File: docs/api-client/README.md (hunk 1 above)
Comment (Jdville03) [reply docs/api-client/README.md]: Sure we can remove for now.

File: docs/api-client/README.md (hunk 3 above)
Comment (Jdville03) [reply docs/api-client/README.md]: Agreed - removing for now.

File: docs/api-client/README.md (hunk 6 above)
Comment (Jdville03) [reply docs/api-client/README.md]: Per discussion during the standup today, i'll include examples for everyone's review too.

Comment (outkaj) [reply docs/api-client/README.md]: Sounds great, @Jdville03! Feel free to reach out if you need anything there.

File: docs/api-client/README.md [hunk 8]
```diff
@@ -24,8 +24,8 @@ GET /template/raw/:title
```
Comment (outkaj) [inline docs/api-client/README.md]: we can set up to run `npm run format` via CI as per discussion with @mroth . i'll make an issue for that

**Pull request merged**